docs/reference/search/search-your-data/retrievers_examples.asciidoc

```diff
@@ -0,0 +1,312 @@
+                   ],
+                   "rank_window_size": 10,
+                   "rank_constant": 1
+               }
+           },
+           "field": "text",
+           "inference_id": "my-awesome-rerank-model",
+           "inference_text": "What are the state of the art applications of AI in information retrieval?"
+       }
+   },
+   "_source": ["text", "topic"]
+}
+
+----
+//NOTCONSOLE
+end::text-similarity-reranker-on-top-of-rrf[]
+
+tag::chaining-text-similarity-reranker-retrievers[]
+[discrete]
+=== Example: Chaining multiple semantic rerankers
+
+Full composability, means that we can also chain together multiple retrievers of the same type. Say that we have another
+very computationally expensive reranker that is more fine-grained for AI content. We can now also rerank the results of a `text_similarity_reranker`,
+using another `text_similarity_reranker` retriever, which could operate on different fields and/or inference services!
+
+[source,js]
+----
+GET retrievers_example/_search
+{
+   "retriever": {
+       "text_similarity_reranker": {
+           "retriever": {
+               "text_similarity_reranker": {
+                   "retriever": {
+                       "knn": {
+                           "field": "vector",
+                           "query_vector": [
+                               0.23,
+                               0.67,
+                               0.89
+                           ],
+                           "k": 3,
+                           "num_candidates": 5
+                       }
+                   },
+                   "rank_window_size": 100,
+                   "field": "text",
+                   "inference_id": "my-awesome-rerank-model",
+                   "inference_text": "What are the state of the art applications of AI in information retrieval?"
+               }
+           },
+           "rank_window_size": 10,
+           "field": "text",
+           "inference_id": "my-other-more-expensive-rerank-model",
+           "inference_text": "Applications of Large Language Models in technology and their impact on user satisfaction"
+       }
+   },
+   "_source": [
+       "text",
+       "topic"
+   ]
+}
+----
+//NOTCONSOLE
+
+
+Note that in the example above, we initially rerank the top 100 documents from the `knn` search using the `my-awesome-rerank-model` reranker,
+and then pick the top 10 results and rerank them using the more fine-grained `my-other-more-expensive-rerank-model`.
+
+end::chaining-text-similarity-reranker-retrievers[]
+
+tag::rrf-retriever-with-aggs[]
+[discrete]
+=== Example: Combine RRF with aggregations
+
+We have seen some examples with retrievers' composability, but we can also now support most of the standard search functionality!
+Let's say that we want to compute aggregations for the `rrf` retriever. Note that the aggregations
+in a compound retriever will be computed based on the nested retrievers it holds. So this means that for the following query
```

Comment:
```suggestion
Retrievers support both composability and most of the standard `_search` functionality. For instance,
we can compute aggregations with the `rrf` retriever. When using a compound retriever,
the aggregations are computed based on its nested retrievers. Here's an example:
```

Reply:
this text seemed to end abruptly, consider adding any missing context!

Reply:
yeap, missed this when I was moving content around. added in c11c47f8e7373e316f029d52a4dbf4df5a7905fa